packages/core/core/src/dumpGraphToGraphViz.js

```diff
@@ -0,0 +1,116 @@
+// @flow
+
+import type {Environment} from '@parcel/types';
+
+import type Graph from './Graph';
+import type {AssetGraphNode, BundleGraphNode} from './types';
+
+import invariant from 'assert';
+import nullthrows from 'nullthrows';
+import path from 'path';
+import BundleGraph from './BundleGraph';
+
+const COLORS = {
+  root: 'gray',
+  asset: 'green',
+  dependency: 'orange',
+  transformer_request: 'cyan',
+  file: 'gray',
+  default: 'white'
+};
+
+export default async function dumpGraphToGraphViz(
+  // $FlowFixMe
+  graph: Graph<AssetGraphNode> | Graph<BundleGraphNode>,
+  name: string
+): Promise<void> {
+  if (
+    process.env.PARCEL_BUILD_ENV === 'production' ||
+    process.env.PARCEL_DUMP_GRAPHVIZ == null
+  ) {
+    return;
```

Comment:
Is DCE good enough to compile out code following an early return? Might need wrap the following code in an `if` with inverted logic.